.github/workflows/ci.yml

```diff
@@ -36,10 +36,37 @@ jobs:
         go install golang.org/x/tools/cmd/goimports@latest
         go install github.com/mattn/goveralls@latest
     - if: ${{ runner.os == 'Linux' }}
-      run: if [ $(goimports -l .) ]; then goimports -d .; echo 'Failed the goimports format check. Please format the code using "goimports -w ."'; exit 1; fi
-    - run: go test -mod=mod -coverpkg="./..." -coverprofile=covprofile ./...
-    - env:
+      env:
         COVERALLS_TOKEN: ${{ secrets.github_token }}
-      run: goveralls -coverprofile=covprofile -service=github -ignore="confgenerator/filter/internal/generated/*/*"
+      run: |
+        if [ $(goimports -l .) ]; then goimports -d .; echo 'Failed the goimports format check. Please format the code using "goimports -w ."'; exit 1; fi
+        go test -mod=mod -coverpkg="./..." -coverprofile=covprofile ./...
+        goveralls -coverprofile=covprofile -service=github -ignore="confgenerator/filter/internal/generated/*/*"
+    - id: 'set-coverage-vars'
+      if: ${{ runner.os == 'Linux' }}
+      name: 'Set Coverage Env Vars'
+      run: echo "ABS_DIR=ng3-customer-telemetry-coverage/absolute/cloud-ops-agent/${{ github.run_id }}" >> "$GITHUB_ENV"
```

Comment:
Hmm, why couldn't this be done with the regular `env`? As I understand it, `github.run_id` is per-workflow, and is not step-dependent, so it could even be set at the workflow level…

Reply:
regular env doesn't support string manipulation

Reply:
I'm confused — you don't seem to have string manipulation here either. If you mean "string _interpolation_" (via `${{…}}`), then regular env does support that. Am I missing something?

Reply:
I am concatenating strings, you can't do that with `env`

Reply:
https://github.com/orgs/community/discussions/26179

Reply:
I see. I suppose something like the following would have worked as well:
```
env:
  COV_DIR=ng3-customer-telemetry-coverage/absolute/cloud-ops-agent/
  COV_SUBDIR=${{ github.run_id }}
…
steps:
- run: |
     mkdir -p "${{ env.COV_DIR }}${{ env.COV_SUBDIR }}"
     echo '…' > ${{ env.COV_DIR }}${{ env.COV_SUBDIR }}/metadata.json
     …
```
but repeating that throughout is fairly cumbersome and error-prone, so using an extra step seems reasonable.